### DISCUSSION TITLE
Issue with email signup and phone signup on Flutter

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#2803] [Questions] [answered]
I had this issue when I was using email as the authentication method , here is the discussion I started about that - https://github.com/supabase/supabase/discussions/2752. The issue in short, I was getting Internal server error when clicking the **Confirm Your Email** link sent to my email. 

Then the supabase team updated their flutter_package to support phone authentication. I totally got rid of the email authentication and jumped into phone authentication method which is what I wanted since the beginning. Now I am having the same issue with phone authentication when I call the verifyOTP method.

Here is my code to signup and to verify the OTP -

final response = await supabase.auth.signUpWithPhone(phoneNumber, '123456');
final response = await supabase.auth.verifyOTP(phoneNumber, otpCode);

I see the user being created inside of Supabase which says waiting for verification and then when I call the verifyOTP method the response on the verifyOTP method returns **Internal server error**.

Please fix this issue at the earliest as possible, as I had been stuck on this for a few days now. Not sure its the problem with my account or with Supabase systems.

Please advise!

**Comment by dshukertjr** [comment, marked as answer]
Hi @sumchans!

I could not reproduce this issue as I was able to run `verifyOTP` without any errors.

Is the error same as last time?:
```
{"code":500,"msg":"Internal server error","error_id":"bf95ca5f-058d-4e41-b03b-482bef80cd5c"}
```
If not, it would be great if you could share the error message!

 Do you have an actual code that I can take a look at?

**Reply by sumchans** [reply]
It doesn't give me all the other details on the iOS simulator, but print response.error says Internal Server error. Please have a look at it over here https://github.com/sumchans/bol_client . It is mostly the same code from Supabase_flutter tutorial, I have just made some changes to the login_page.dart file. I have justchanged the authentication from email to phone. This is not letting me continue for sometime. I have no idea if it is my issue or Supabases issue. Please let me know.

**Reply by dshukertjr** [reply]
@sumchans 
I really like the update you did on the quick start app! Amazing work!

I was actually able to do verify my OTP without any issue with your code. If you could provide the project ID of your Supabase instance, maybe the team can look into the logs on the server to see what is going on here.

You can find your project ID from your Supabase console's URL!

**Reply by sumchans** [reply]
Thats Weird, I believe this is the one you are looking for. Thats what I always thought, it could be something related to my project.
Project Id - https://bxqfkhvqwnahgzxczlsd.supabase.co

**Reply by sumchans** [reply]
I did spin up another project in Supabase, and tried my code there. it works. Maybe I have to restart my flutter project with the new Supabase project. This issue might pop up for other users as well, wish to know what's going on with my initial project.

**Reply by sumchans** [reply]
I am still in the beginning phase of my development. So right now the only option to make it faster, was to create a new project and continue working on that one. Don't worry about the other project for now, I had to delete it to keep the same name. We can put this issue to the side and will let the team know if this happens in my new project.

**Reply by dshukertjr** [reply]
@sumchans So sorry for the inconvenience, but it's so great to hear that you were able to make it work by creating another project!

**Reply by dshukertjr** [reply]
Also, you can view the auth related logs here at https://app.supabase.io/project/YOUR_PROJECT_ID/auth/settings so if you ever face something similar, maybe you can find out more of what is causing the issue there! With a more detailed log, we might be able to better track down the specific issue and fix it sooner!

**Reply by sumchans** [reply]
Oh okay, sure will keep this in mind. Thanks Tyler.

**Comment by J0** [comment]
Hey all,

Thanks for the query! As @dshukertjr mentioned, do check out the logs for easier debugging -- we've now added a built in [log explorer](https://supabase.com/docs/guides/platform/logs) for easier log tracking.

Joel
